modules/confiantRtdProvider.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * This module provides comprehensive detection of security, quality, and privacy threats by Confiant Inc,
+ * the industry leader in real-time detecting and blocking of bad ads
+ *
+ * The {@link module:modules/realTimeData} module is required
+ * The module will inject a Confiant Inc. script into the page to monitor ad impressions
+ * @module modules/cleanioRtdProvider
+ * @requires module:modules/realTimeData
+ */
+
+import { submodule } from '../src/hook.js';
+import { logError, insertElement, generateUUID } from '../src/utils.js';
+import * as events from '../src/events.js';
+import CONSTANTS from '../src/constants.json';
+
+/**
+ * Injects the Confiant Inc. configuration script into the page, based on proprtyId provided
+ * @param {string} propertyId
+ */
+function injectConfigScript(propertyId) {
+  const scriptElement = document.createElement('script');
+  scriptElement.type = 'text/javascript';
+  scriptElement.src = `https://cdn.confiant-integrations.net/${propertyId}/gpt_and_prebid/config.js`;
+
+  insertElement(scriptElement);
+}
+
+/**
+ * Set up page with Confiant integration
+ * @param {Object} config
+ */
+function setupPage(config) {
+  const propertyId = config?.params?.propertyId;
+  if (!propertyId) {
+    logError('Confiant pbjs module: no propertyId provided');
+    return false;
+  }
+
+  if (window.frames['cnftComm']) {
+    subscribeToConfiantCommFrame(window);
+  } else {
+    setUpMutationObserver();
+  }
+  injectConfigScript(propertyId);
+  return true;
+}
+
+/**
+ * Subscribe to window's message events to report Billable events
+ * @param {Window} targetWindow window instance to subscribe to
+ */
+function subscribeToConfiantCommFrame(targetWindow) {
+  targetWindow.addEventListener('message', reportBillableEvents);
```

Comment:
add window.confiant.settings.isMGBL = true
to enable billable events

Reply:
if there are multiple confiant stacks on the page we might need to enable for the appropriate config object based on the property id